cmd: Honor repo location change from openshift/os

peterbaouoft opened the pull request:
As mentioned in https://github.com/openshift/os/issues/157#issuecomment-403062182,
this patch aims to change the repo location so that it will be matching
the one in openshift/os.

**Note**: Has not tested this yet... I need to learn a bit about how to build golang binaries + `rsync or scp` it into Atomic Host for testing... Will remove WIP once I have tested it, (should be by eod today :) )

ashcrow commented:
Thanks @peterbaouoft! Let me know when to review!

peterbaouoft commented:
> Let me know when to review!

:D, will do , thanks :). 

I have a quick question about getting the binary of pivot actually. Wanted to ask, what are the steps for building this upstream? I did `make install` and `make build`, but got the following error:
```
[root@localhost pivot]# make install
rm -f pivot
go build -ldflags '-X main.version=0.0.0 -X main.commitHash=006c71a8b9fa78380620fb1c00fc61dc06c0b6be -X main.buildTime=1530899832' -o pivot main.go
main.go:11:2: cannot find package "github.com/ashcrow/pivot/cmd" in any of:
	/usr/lib/golang/src/github.com/ashcrow/pivot/cmd (from $GOROOT)
	/root/go/src/github.com/ashcrow/pivot/cmd (from $GOPATH)
main.go:8:2: cannot find package "github.com/spf13/pflag" in any of:
	/usr/lib/golang/src/github.com/spf13/pflag (from $GOROOT)
	/root/go/src/github.com/spf13/pflag (from $GOPATH)
make: *** [Makefile:32: build] Error 1
```

ashcrow commented:
@peterbaouoft you need to check the code out in your `GOROOT` and make sure it's in `src/github.com/ashcrow/pivot/`

peterbaouoft commented:
> you need to check the code out in your GOROOT and make sure it's in src/github.com/ashcrow/pivot/

Yup, makes sense, tried it and it worked \o/.

peterbaouoft commented:
Tested this change on `28.20180527.0` fedora atomic host vagrant box. And it worked. FYI: the log is as follows (`pivot` binary is `scp`ed into the host). And dropping WIP @ashcrow.

```
[root@localhost ~]# ./pivot registry.svc.ci.openshift.org/rhcos/os:latest
pivot version 0.0.0 (9678a1aedd17715f25f0fae45ddc2bc15dd8dc2c)
Running: skopeo inspect docker://registry.svc.ci.openshift.org/rhcos/os:latest
03b98184dc2939e423efefaa044d31ed06d717ee09351affab5b68ed0b7562c6
Running: podman run -d --name ostree-container-pivot --entrypoint sleep registry.svc.ci.openshift.org/rhcos/os:latest@sha256:05815042ae65eabac9c1406b7a8cc8fae45345407fe95cc939367f2d2a031ea2 infinity
Running: podman mount 7e4899ca619dcdc3f8b21c0c703716360bb2640add1e98f3f38fc400aa7cdd99
Running: ostree --repo=srv/repo refs
Running: ostree --repo=srv/repo rev-parse openshift/3.10/x86_64/os
1262 metadata, 6386 content objects imported                                                                                                                                                                       
Copying /etc changes: 22 modified, 0 removed, 62 added
Transaction complete; bootconfig swap: yes; deployment count change: 0
Freed: 949.1 MB (pkgcache branches: 0)
Upgraded:
  GeoIP 1.5.0-11.el7 -> 1.5.0-12.el7
  NetworkManager 1:1.10.2-14.el7_5 -> 1:1.12.0-0.1.el7
  NetworkManager-libnm 1:1.10.2-14.el7_5 -> 1:1.12.0-0.1.el7
  atomic-registries 1:1.22.1-3.git2fd0860.el7 -> 1:1.22.1-22.git5a342e3.el7
  audit-libs 2.8.1-3.el7 -> 2.8.4-1.el7
  audit-libs-python 2.8.1-3.el7 -> 2.8.4-1.el7
  bash 4.2.46-30.el7 -> 4.2.46-31.el7
  bind-libs 32:9.9.4-61.el7 -> 32:9.9.4-67.el7
  bind-libs-lite 32:9.9.4-61.el7 -> 32:9.9.4-67.el7
  bind-license 32:9.9.4-61.el7 -> 32:9.9.4-67.el7
  bind-utils 32:9.9.4-61.el7 -> 32:9.9.4-67.el7
  binutils 2.27-27.base.el7 -> 2.27-34.base.el7
  buildah 0.16.0-2.git6f7d05b.el7 -> 1.1-1.gitfbf46d3.el7
  ceph-common 1:0.94.5-2.el7 -> 1:10.2.5-4.el7
  checkpolicy 2.5-6.el7 -> 2.5-7.el7
  container-selinux 2:2.55-1.el7 -> 2:2.66-1.el7
  container-storage-setup 0.9.0-1.rhel75.gite0997c3.el7 -> 0.10.0-1.gitdf0dcd5.el7
  coreutils 8.22-21.el7 -> 8.22-23.el7
  cri-tools 1.0.0-5.gitf6ed14e.el3 -> 1.0.0.beta.0.21-2e22a757466719d43cc76aab0e01019081f8c27a.a5b6369696fd69fcb7242837428dd2f836cbc2df.el3
  criu 3.5-4.el7 -> 3.9-4.el7
  cryptsetup 1.7.4-4.el7 -> 2.0.3-1.el7
  cryptsetup-libs 1.7.4-4.el7 -> 2.0.3-1.el7
  curl 7.29.0-46.el7 -> 7.29.0-50.el7
  device-mapper 7:1.02.146-4.el7 -> 7:1.02.148-1.el7
  device-mapper-event 7:1.02.146-4.el7 -> 7:1.02.148-1.el7
  device-mapper-event-libs 7:1.02.146-4.el7 -> 7:1.02.148-1.el7
  device-mapper-libs 7:1.02.146-4.el7 -> 7:1.02.148-1.el7
  device-mapper-multipath 0.4.9-119.el7 -> 0.4.9-120.el7
  device-mapper-multipath-libs 0.4.9-119.el7 -> 0.4.9-120.el7
  dmidecode 1:3.0-5.el7 -> 1:3.1-2.el7
  dnsmasq 2.76-5.el7 -> 2.76-7.el7
  docker 2:1.13.1-63.git94f4240.el7 -> 2:1.13.1-68.gitdded712.el7
  docker-client 2:1.13.1-63.git94f4240.el7 -> 2:1.13.1-68.gitdded712.el7
  docker-common 2:1.13.1-63.git94f4240.el7 -> 2:1.13.1-68.gitdded712.el7
  docker-lvm-plugin 2:1.13.1-63.git94f4240.el7 -> 2:1.13.1-68.gitdded712.el7
  docker-novolume-plugin 2:1.13.1-63.git94f4240.el7 -> 2:1.13.1-68.gitdded712.el7
  docker-rhel-push-plugin 2:1.13.1-63.git94f4240.el7 -> 2:1.13.1-68.gitdded712.el7
  dracut 033-535.el7 -> 033-547.el7
  dracut-network 033-535.el7 -> 033-547.el7
  efibootmgr 15-2.el7 -> 17-1.el7
  efivar-libs 31-4.el7 -> 36-6.el7
  elfutils-default-yama-scope 0.170-4.el7 -> 0.172-2.el7
  elfutils-libelf 0.170-4.el7 -> 0.172-2.el7
  elfutils-libs 0.170-4.el7 -> 0.172-2.el7
  ethtool 2:4.8-7.el7 -> 2:4.8-8.el7
  file 5.11-33.el7 -> 5.11-35.el7
  file-libs 5.11-33.el7 -> 5.11-35.el7
  findutils 1:4.5.11-5.el7 -> 1:4.5.11-6.el7
  freetype 2.4.11-15.el7 -> 2.8-12.el7
  gdisk 0.8.6-5.el7 -> 0.8.10-2.el7
  glib2 2.54.2-2.el7 -> 2.56.1-1.el7
  glibc 2.17-222.el7 -> 2.17-259.el7
  glibc-common 2.17-222.el7 -> 2.17-259.el7
  glusterfs 3.8.4-53.el7 -> 3.12.2-12.el7
  glusterfs-client-xlators 3.8.4-53.el7 -> 3.12.2-12.el7
  glusterfs-fuse 3.8.4-53.el7 -> 3.12.2-12.el7
  glusterfs-libs 3.8.4-53.el7 -> 3.12.2-12.el7
  gobject-introspection 1.50.0-1.el7 -> 1.56.1-1.el7
  grub2 1:2.02-0.44.el7 -> 1:2.02-0.65.el7_4.2
  grub2-tools 1:2.02-0.44.el7 -> 1:2.02-0.65.el7_4.2
  gssproxy 0.7.0-17.el7 -> 0.7.0-21.el7
  initscripts 9.49.41-1.el7 -> 9.49.44-1.el7
  iptables 1.4.21-24.1.el7_5 -> 1.4.21-28.el7
  iptables-services 1.4.21-24.1.el7_5 -> 1.4.21-28.el7
  json-glib 1.2.6-1.el7 -> 1.4.2-1.el7
  kernel 3.10.0-862.2.3.el7 -> 3.10.0-907.el7
  kmod 20-21.el7 -> 20-22.el7
  kmod-libs 20-21.el7 -> 20-22.el7
  kpartx 0.4.9-119.el7 -> 0.4.9-120.el7
  krb5-libs 1.15.1-19.el7 -> 1.15.1-32.el7
  libbasicobjects 0.1.1-29.el7 -> 0.1.1-32.el7
  libblkid 2.23.2-52.el7 -> 2.23.2-57.el7
  libcgroup 0.41-15.el7 -> 0.41-19.el7
  libcollection 0.7.0-29.el7 -> 0.7.0-32.el7
  libcroco 0.6.11-1.el7 -> 0.6.12-4.el7
  libcurl 7.29.0-46.el7 -> 7.29.0-50.el7
  libdhash 0.5.0-29.el7 -> 0.5.0-32.el7
  libgcc 4.8.5-28.el7_5.1 -> 4.8.5-36.el7
  libgomp 4.8.5-28.el7_5.1 -> 4.8.5-36.el7
  libini_config 1.3.1-29.el7 -> 1.3.1-32.el7
  libipa_hbac 1.16.0-19.el7 -> 1.16.2-1.el7
  libldb 1.2.2-1.el7 -> 1.3.3-1.el7
  libmount 2.23.2-52.el7 -> 2.23.2-57.el7
  libpath_utils 0.2.1-29.el7 -> 0.2.1-32.el7
  librados2 1:0.94.5-2.el7 -> 1:10.2.5-4.el7
  librbd1 1:0.94.5-2.el7 -> 1:10.2.5-4.el7
  libref_array 0.1.5-29.el7 -> 0.1.5-32.el7
  libselinux 2.5-12.el7 -> 2.5-13.el7
  libselinux-python 2.5-12.el7 -> 2.5-13.el7
  libselinux-utils 2.5-12.el7 -> 2.5-13.el7
  libsemanage 2.5-11.el7 -> 2.5-12.el7
  libsemanage-python 2.5-11.el7 -> 2.5-12.el7
  libsepol 2.5-8.1.el7 -> 2.5-9.el7
  libsmbclient 4.7.1-6.el7 -> 4.8.2-1.el7
  libsolv 0.6.27-1.el7.centos -> 0.6.34-2.el7
  libssh2 1.4.3-10.el7_2.1 -> 1.4.3-12.el7
  libsss_autofs 1.16.0-19.el7 -> 1.16.2-1.el7
  libsss_certmap 1.16.0-19.el7 -> 1.16.2-1.el7
  libsss_idmap 1.16.0-19.el7 -> 1.16.2-1.el7
  libsss_nss_idmap 1.16.0-19.el7 -> 1.16.2-1.el7
  libsss_sudo 1.16.0-19.el7 -> 1.16.2-1.el7
  libstdc++ 4.8.5-28.el7_5.1 -> 4.8.5-36.el7
  libtalloc 2.1.10-1.el7 -> 2.1.13-1.el7
  libtevent 0.9.33-2.el7 -> 0.9.36-1.el7
  libuuid 2.23.2-52.el7 -> 2.23.2-57.el7
  libwbclient 4.7.1-6.el7 -> 4.8.2-1.el7
  linux-firmware 20180220-62.git6d51311.el7 -> 20180529-65.git7518922.el7
  logrotate 3.8.6-15.el7 -> 3.8.6-17.el7
  lvm2 7:2.02.177-4.el7 -> 7:2.02.179-1.el7
  lvm2-libs 7:2.02.177-4.el7 -> 7:2.02.179-1.el7
  microcode_ctl 2:2.1-29.el7 -> 2:2.1-33.el7
  mokutil 0.9-2.el7 -> 12-1.el7
  net-tools 2.0-0.22.20131004git.el7 -> 2.0-0.24.20131004git.el7
  nmap-ncat 2:6.40-13.el7 -> 2:6.40-16.el7
  oci-systemd-hook 1:0.1.15-2.gitc04483d.el7 -> 1:0.1.16-1.git05bd9a0.el7
  openldap 2.4.44-15.el7_5 -> 2.4.44-16.el7
  openssl 1:1.0.2k-12.el7 -> 1:1.0.2k-15.el7
  openssl-libs 1:1.0.2k-12.el7 -> 1:1.0.2k-15.el7
  podman 0.4.1-4.gitb51d327.el7 -> 0.6.1-3.git3e0ff12.el7
  policycoreutils 2.5-22.el7 -> 2.5-23.el7
  policycoreutils-python 2.5-22.el7 -> 2.5-23.el7
  polkit 0.112-14.el7 -> 0.112-15.el7
  procps-ng 3.3.10-17.el7 -> 3.3.10-23.el7
  python 2.7.5-68.el7 -> 2.7.5-74.el7
  python-ethtool 0.8-5.el7 -> 0.8-7.el7
  python-libs 2.7.5-68.el7 -> 2.7.5-74.el7
  python-linux-procfs 0.4.9-3.el7 -> 0.4.9-4.el7
  python-perf 3.10.0-862.2.3.el7 -> 3.10.0-907.el7
  python-rados 1:0.94.5-2.el7 -> 1:10.2.5-4.el7
  python-rbd 1:0.94.5-2.el7 -> 1:10.2.5-4.el7
  python-sssdconfig 1.16.0-19.el7 -> 1.16.2-1.el7
  python-urlgrabber 3.10-8.el7 -> 3.10-9.el7
  rpm 4.11.3-32.el7 -> 4.11.3-35.el7
  rpm-build-libs 4.11.3-32.el7 -> 4.11.3-35.el7
  rpm-libs 4.11.3-32.el7 -> 4.11.3-35.el7
  rpm-python 4.11.3-32.el7 -> 4.11.3-35.el7
  samba-client-libs 4.7.1-6.el7 -> 4.8.2-1.el7
  samba-common 4.7.1-6.el7 -> 4.8.2-1.el7
  samba-common-libs 4.7.1-6.el7 -> 4.8.2-1.el7
  selinux-policy 3.13.1-192.el7_5.3 -> 3.13.1-204.el7
  selinux-policy-targeted 3.13.1-192.el7_5.3 -> 3.13.1-204.el7
  setools-console 3.3.8-2.el7 -> 3.3.8-3.el7
  setools-libs 3.3.8-2.el7 -> 3.3.8-3.el7
  setup 2.8.71-9.el7 -> 2.8.71-10.el7
  sg3_utils 1.37-12.el7 -> 1.37-15.el7
  sg3_utils-libs 1.37-12.el7 -> 1.37-15.el7
  shadow-utils 2:4.1.5.1-24.el7 -> 2:4.1.5.1-25.el7
  skopeo 1:0.1.29-3.dev.git7add6fc.el7 -> 1:0.1.30-1.dev.gitca3bff6.el7
  skopeo-containers 1:0.1.29-3.dev.git7add6fc.el7 -> 1:0.1.30-1.dev.gitca3bff6.el7
  sssd 1.16.0-19.el7 -> 1.16.2-1.el7
  sssd-ad 1.16.0-19.el7 -> 1.16.2-1.el7
  sssd-client 1.16.0-19.el7 -> 1.16.2-1.el7
  sssd-common 1.16.0-19.el7 -> 1.16.2-1.el7
  sssd-common-pac 1.16.0-19.el7 -> 1.16.2-1.el7
  sssd-ipa 1.16.0-19.el7 -> 1.16.2-1.el7
  sssd-krb5 1.16.0-19.el7 -> 1.16.2-1.el7
  sssd-krb5-common 1.16.0-19.el7 -> 1.16.2-1.el7
  sssd-ldap 1.16.0-19.el7 -> 1.16.2-1.el7
  sssd-proxy 1.16.0-19.el7 -> 1.16.2-1.el7
  strace 4.12-6.el7 -> 4.12-7.el7
  subscription-manager 1.20.11-1.el7_5 -> 1.21.5-4.el7
  subscription-manager-plugin-container 1.20.11-1.el7_5 -> 1.21.5-4.el7
  subscription-manager-plugin-ostree 1.20.11-1.el7_5 -> 1.21.5-4.el7
  subscription-manager-rhsm 1.20.11-1.el7_5 -> 1.21.5-4.el7
  subscription-manager-rhsm-certificates 1.20.11-1.el7_5 -> 1.21.5-4.el7
  systemd 219-57.el7 -> 219-58.el7
  systemd-libs 219-57.el7 -> 219-58.el7
  systemd-sysv 219-57.el7 -> 219-58.el7
  tuned 2.9.0-1.el7 -> 2.10.0-0.1.rc1.el7
  tuned-profiles-atomic 2.9.0-1.el7 -> 2.10.0-0.1.rc1.el7
  tzdata 2018d-1.el7 -> 2018e-3.el7
  util-linux 2.23.2-52.el7 -> 2.23.2-57.el7
  wpa_supplicant 1:2.6-9.el7 -> 1:2.6-11.el7
Downgraded:
  conmon 1.11.0-1.git441bd3d.el3 -> 1.10.4.6-8bb8b9dfce12ac67fcd6e1d7d29a47f4fbc6fab9.79813fdbb90db37f3d4fb52f7a1232500d3d1a29.el3
  conntrack-tools 1.4.4-3.el7_3 -> 1.4.4-3.el7
  cri-o 1.11.0-1.git441bd3d.el3 -> 1.10.4.6-8bb8b9dfce12ac67fcd6e1d7d29a47f4fbc6fab9.79813fdbb90db37f3d4fb52f7a1232500d3d1a29.el3
  dbus 1:1.10.24-9.el7 -> 1:1.10.24-7.el7
  dbus-libs 1:1.10.24-9.el7 -> 1:1.10.24-7.el7
  ignition 0.26.0-0.4.git7610725.el7 -> 0.26.0-0.2.git7610725.el7
  ignition-dracut 0.26.0-0.4.git7610725.el7 -> 0.26.0-0.2.git7610725.el7
  origin 3.11.0-0.alpha.0.245.00db30d -> 3.11.0-0.alpha.0.68.e004e65
  origin-clients 3.11.0-0.alpha.0.245.00db30d -> 3.11.0-0.alpha.0.68.e004e65
  origin-hyperkube 3.11.0-0.alpha.0.245.00db30d -> 3.11.0-0.alpha.0.68.e004e65
  origin-node 3.11.0-0.alpha.0.245.00db30d -> 3.11.0-0.alpha.0.68.e004e65
  ostree 2018.6.34-10c2fc33f68fbc7d8b9d29888d76cc0b44a170be.61448e46ae20d64b69b7d5b0d952fdee94aa8363.el7.centos -> 2018.6.16-d686056254294fcabff472e71ebe486ec82e070e.56ead4746a7f3a87b295b4720128a977e7c10eae.el7.centos
  ostree-grub2 2018.6.34-10c2fc33f68fbc7d8b9d29888d76cc0b44a170be.61448e46ae20d64b69b7d5b0d952fdee94aa8363.el7.centos -> 2018.6.16-d686056254294fcabff472e71ebe486ec82e070e.56ead4746a7f3a87b295b4720128a977e7c10eae.el7.centos
  ostree-libs 2018.6.34-10c2fc33f68fbc7d8b9d29888d76cc0b44a170be.61448e46ae20d64b69b7d5b0d952fdee94aa8363.el7.centos -> 2018.6.16-d686056254294fcabff472e71ebe486ec82e070e.56ead4746a7f3a87b295b4720128a977e7c10eae.el7.centos
  redhat-release-coreos 0-eef71a43521ea22f2cffdaa1509788a2fa8401a9.el3 -> 0-b6b1308340c42162e7871799381305c57eba36fc.el3
  rpm-ostree 2018.6.9-2e237513b08aceb69f2cf2095eebe0d5211d5c19.f2b60ed3536e1022ccb94a721a0cccc00635a5a9.el7.centos -> 2018.5.60-07ecfca3d9fcf5012007f53396d43a002b3e8cdf.48080d8ef7b35dd36c3f601d06e7c9fc4d54e1e1.el7.centos
  rpm-ostree-libs 2018.6.9-2e237513b08aceb69f2cf2095eebe0d5211d5c19.f2b60ed3536e1022ccb94a721a0cccc00635a5a9.el7.centos -> 2018.5.60-07ecfca3d9fcf5012007f53396d43a002b3e8cdf.48080d8ef7b35dd36c3f601d06e7c9fc4d54e1e1.el7.centos
Removed:
  grub2-efi-1:2.02-0.44.el7.x86_64
  redhat-logos-70.0.3-6.el7.noarch
  shim-0.9-2.el7.x86_64
Added:
  boost-iostreams-1.53.0-27.el7.x86_64
  boost-program-options-1.53.0-27.el7.x86_64
  boost-random-1.53.0-27.el7.x86_64
  boost-regex-1.53.0-27.el7.x86_64
  grub2-common-1:2.02-0.65.el7_4.2.noarch
  grub2-efi-x64-1:2.02-0.65.el7_4.2.x86_64
  grub2-pc-1:2.02-0.65.el7_4.2.x86_64
  grub2-pc-modules-1:2.02-0.65.el7_4.2.noarch
  grub2-tools-extra-1:2.02-0.65.el7_4.2.x86_64
  grub2-tools-minimal-1:2.02-0.65.el7_4.2.x86_64
  json-c-0.11-4.el7_0.x86_64
  libpng-2:1.5.13-7.el7_2.x86_64
  libsmartcols-2.23.2-57.el7.x86_64
  python-kitchen-1.1.1-5.el7.noarch
  shim-x64-12-1.el7.x86_64
Run "systemctl reboot" to start a reboot
03b98184dc2939e423efefaa044d31ed06d717ee09351affab5b68ed0b7562c6
[root@localhost ~]# systemctl reboot 
Connection to 192.168.121.112 closed by remote host.
Connection to 192.168.121.112 closed.
[rbao@unused-10-15-17-147 test_ah]$ vagrant ssh
Last login: Fri Jul  6 18:25:08 2018 from 192.168.121.1
Red Hat CoreOS
  Information: https://url.corp.redhat.com/redhat-coreos
  Bugs: https://github.com/openshift/os
---
```

peterbaouoft commented:
Thanks for reviewing!
